state-sync/storage-service/server/src/lib.rs

```diff
@@ -956,29 +956,30 @@ impl StorageReader {
         latest_version: Version,
         transactions_range: &Option<CompleteDataRange<Version>>,
     ) -> Result<Option<CompleteDataRange<Version>>, Error> {
+        let pruner_enabled = self.storage.is_state_pruner_enabled().map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
```

Comment:
We now know the real range of available state trees (even regardless if the pruner is enabled), shall we just return the real thing? @JoshLind

(and `is_state_pruner_enabled()` isn't useful in the case)

Reply:
Discussed offline. We can do it in another PR.